sample-apps/javascript-sample-app/server.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS'" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+'use strict'
+
+const sdk = require("./common");
+const Worker = require("worker_threads");
+const http = require('http');
+const AWS = require('aws-sdk');
+const fetch = require ("node-fetch");
+
+// tracer
+const api = require('@opentelemetry/api'); 
+const common_span_attributes = { signal: 'trace', language: 'javascript' };
+
+// config
+const create_cfg = require('./config');
+const cfg = create_cfg.create_config('./config.yaml');
+
+
+// initialise sdk (metric and trace provider) and start server and a separate thread for random-metrics generation.
+sdk.nodeSDKBuilder()
+    .then(() => {
+    startServer();
+    const worker = new Worker.Worker('./random-metrics.js'); 
+});
+
+function startServer() {
+    const server = http.createServer(handleRequest);
+    server.listen(cfg.Port, cfg.Host, (err) => {
+        if (err) {
+            throw err;
+        }
+        console.log(`Node HTTP listening on ${cfg.Host}:${cfg.Port}`);
+    });
+}
+
+function handleRequest(req, res) {
+    const { updateTotalBytesSent, updateLatencyTime, updateApiRequestsMetric } = require('./request-metrics');
+    const requestStartTime = new Date().getMilliseconds();
+    const routeMapper = {
+        '/': (req, res) => {
+            res.end('OK.');
+        },
+        '/aws-sdk-call': (req, res) => {
+            const s3 = new AWS.S3();
```

Comment:
can you create functions for each request handler instead of populating them here?

Reply:
Right now creating a separate function is giving error on res.end(). I am looking for a solution.

Reply:
Hi. Thanks for the help on this. Updated now.